service/history/events/cache.go

```diff
@@ -120,12 +121,24 @@ func newEventKey(namespaceID, workflowID, runID string, eventID int64) eventKey
 	}
 }
 
+func (e *CacheImpl) validateIds(namespaceID, workflowID, runID string, eventID int64) {
+	if len(namespaceID) == 0 || len(workflowID) == 0 || len(runID) == 0 || eventID < common.FirstEventID {
+		// This is definitely a bug, but just warn and don't crash so we can find anywhere this happens.
+		e.logger.Warn("one or more ids is invalid in event cache",
+			tag.WorkflowID(workflowID),
+			tag.WorkflowRunID(runID),
+			tag.WorkflowNamespaceID(namespaceID),
+			tag.WorkflowEventID(eventID))
+	}
+}
+
 func (e *CacheImpl) GetEvent(namespaceID, workflowID, runID string, firstEventID, eventID int64,
 	branchToken []byte) (*historypb.HistoryEvent, error) {
 	e.metricsClient.IncCounter(metrics.EventsCacheGetEventScope, metrics.CacheRequests)
 	sw := e.metricsClient.StartTimer(metrics.EventsCacheGetEventScope, metrics.CacheLatency)
 	defer sw.Stop()
 
+	e.validateIds(namespaceID, workflowID, runID, eventID)
```

Comment:
I could move these calls into `newEventKey` if I make that a method on the struct?